draft-ietf-tls-esni.md

```diff
@@ -488,32 +478,56 @@ by the server. (Recall that servers might provide multiple ESNIRecord values in
 to a ESNI record query, each containing an ESNIKeys value.) The client MUST NOT send encrypted SNI using groups or
 cipher suites not advertised by the server.
 
-When offering an encrypted SNI, the client MUST NOT offer to resume any non-ESNI
-PSKs. It additionally MUST NOT offer to resume any sessions for TLS 1.2 or
-below.
+When offering an encrypted SNI, the client MUST NOT offer a session ticket
+obtained through a non-ESNI TLS session. It additionally MUST NOT offer to
+resume any sessions for TLS 1.2 or below.
+
+### Generating ClientHello {#generate-clienthello}
+
+The client then builds the first ClientHello, that has the following properties:
+
+* Contains a "server_name" extension that carries the true SNI DNS name in
+  cleartext.
+* Immediately following the "server_name" extension, contains an
+  "encrypted_server_name" extension carrying the ClientEncryptedSNI structure of
+  type `esni_parameters`.
+
+The client MUST NOT send a "cached_info" extension {{!RFC7924}} with a
+CachedObject entry whose CachedInformationType is "cert".
+
```

Comment:
The text is a bit awkward: the client is not yet sending the Client Hello. How about "The client MUST NOT include a "cached info" extension"?

Also, it is good practice to follow "MUST NOT" with the reason why not. In this case "since this indication would divulge the true value of the SNI".

Reply:
Nice catch!

Did s/send/include/ in daedef9.

Reply:
Clarification regarding why has also gone into https://github.com/tlswg/draft-ietf-tls-esni/pull/181 and merged into this PR.

Reply:
The upstream PR has been merged.